monai/losses/unified_focal_loss.py

```
@@ -20,221 +20,146 @@
-        self.gamma = gamma
-        self.epsilon = epsilon
+        if self.use_softmax:
+            input = torch.nn.functional.softmax(input, dim=1)
 
-    def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor) -> torch.Tensor:
-        n_pred_ch = y_pred.shape[1]
+        n_pred_ch = input.shape[1]
 
         if self.to_onehot_y:
             if n_pred_ch == 1:
                 warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
```

Ruff (0.14.13)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)